packages/react-server/src/ReactServerStreamConfigBrowser.js

```diff
@@ -21,24 +21,64 @@ export function flushBuffered(destination: Destination) {
   // transform streams. https://github.com/whatwg/streams/issues/960
 }
 
+let currentView = null;
+let writtenBytes = 0;
+
 export function beginWriting(destination: Destination) {}
 
 export function writeChunk(
   destination: Destination,
   chunk: PrecomputedChunk | Chunk,
 ): void {
-  destination.enqueue(chunk);
+  if (currentView === null) {
```

Comment:
Why do this lazily here rather than eagerly in `beginWriting`? You can cheat flow and assume that it has already been initialized at this point since we know it has. That way you don't need a runtime check for every call.

Reply:
I forget the original intent but I changed it to match your suggestion and also figured it was more defensive in case endWriting was ever not called (may be impossible but figured a wayward throw might leave this as a possibility)